apps/console/src/features/roles/components/edit-role/edit-role-basic-updated.tsx

```diff
@@ -0,0 +1,266 @@
+import { AxiosResponse } from "axios";
+import React, { FunctionComponent, ReactElement, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useDispatch } from "react-redux";
+import { Dispatch } from "redux";
+import { Divider } from "semantic-ui-react";
+import { AppConstants, history } from "../../../core";
+import { deleteRoleById, searchRoleList, updateRoleDetails } from "../../api";
+import { RoleConstants } from "../../constants";
+import { PatchRoleDataInterface, RoleEditSectionsInterface, SearchRoleInterface } from "../../models/roles";
+
+/**
+ * Interface to contain props needed for component
+ */
+type BasicRoleProps = IdentifiableComponentInterface & RoleEditSectionsInterface;
+
+const FORM_ID: string = "edit-role-basic";
+
+/**
+ * Component to edit basic role details.
+ *
+ * @param props - Role object containing details which needs to be edited.
+ */
+export const BasicRoleDetails: FunctionComponent<BasicRoleProps> = (props: BasicRoleProps): ReactElement => {
+    const { t } = useTranslation();
+    const dispatch: Dispatch = useDispatch();
+
+    const {
+        role,
+        onRoleUpdate,
+        isReadOnly,
+        tabIndex,
+        [ "data-componentid" ]: componentid
+    } = props;
+
+    const [ showRoleDeleteConfirmation, setShowDeleteConfirmationModal ] = useState<boolean>(false);
+    const [ isSubmitting, setIsSubmitting ] = useState<boolean>(false);
+
+    /**
+     * Dispatches the alert object to the redux store.
+     *
+     * @param alert - Alert object.
+     */
+    const handleAlerts = (alert: AlertInterface): void => {
+        dispatch(addAlert(alert));
+    };
+
+    /**
+     * Function which will handle role deletion action.
+     *
+     * @param id - Role ID which needs to be deleted
+     */
+    const handleOnDelete = (): void => {
+        deleteRoleById(role.id).then(() => {
+            handleAlerts({
+                description: t("console:manage.features.roles.notifications.deleteRole.success.description"),
+                level: AlertLevels.SUCCESS,
+                message: t("console:manage.features.roles.notifications.deleteRole.success.message")
+            });
+
+            history.push(AppConstants.getPaths().get("ROLES"));
+        });
+    };
+
+    /**
+     * Validates the Form.
+     *
+     * @param values - Form Values.
+     * @returns Form validation.
+     */
+    const validateForm = async (values: any): Promise<any> => {
+
+        const errors: any = {
+            roleName: undefined
+        };
+
+        if (values.roleName?.toString().trim().length >= 3) {
+
+            const roleName: string = values.roleName?.toString().trim();
+
+            const searchData: SearchRoleInterface = {
+                filter: "displayName eq " + roleName,
+                schemas: [
+                    "urn:ietf:params:scim:api:messages:2.0:SearchRequest"
```

Comment:
Shall we add this to constants?

Reply:
Fixed with https://github.com/wso2/identity-apps/pull/4224/commits/9d912516de37cde05e2b5793fa9765e358586871